test.js

```diff
@@ -163,6 +163,22 @@ test('preserves the original function name', t => {
 	t.is(mem(function foo() {}).name, 'foo'); // eslint-disable-line func-names
 });
 
+test('.decorator()', t => {
+	class TestClass {
+		constructor() {
+			this.i = 0;
+		}
+
+		counter() {
+			return ++this.i;
+		}
+	}
+
+	mem.decorator()(TestClass.prototype.counter, 'counter', TestClass.prototype);
```

Comment:
Whatever we support, we should write exact tests for it. The less real the tests, the less useful they are.

Since the implementations are different (https://github.com/babel/babel/issues/8864), if we support both we should also test both.

For the sake of simplicity (implementation, documentation, and testing), I'd suggest supporting TypeScript’s only.

Reply:
I only want to support the TypeScript implementation.